onnxruntime/core/providers/cuda/math/topk.cc

```diff
@@ -89,7 +89,7 @@ Status TopK<inputk>::ComputeInternal(OpKernelContext* ctx) const {
   auto tensor_V = ctx->Output(0, output_shape);
   auto tensor_I = ctx->Output(1, output_shape);
 
-  if (0 == K_) {
+  if (output_shape.Size() == 0) {
```

Comment:
Potentially spotted another unrelated issue - we are storing K_ (parsed from input tensor) in a mutable member of the class instance - if K were to change across multiple concurrent runs, it is going to cause an issue

Reply:
FIx: https://github.com/microsoft/onnxruntime/pull/19917